test/test_torch.py

```diff
@@ -3293,6 +3299,8 @@ def _generate_noncontiguous_input():
 
             out = np.broadcast_to(np.random.random((1, 4)),
                                   initial_shape)
+            # Note: non-writeable NumPy arrays will warn if converted to tensors
+            out.setflags(write=True)
```

Comment:
just out of curiosity -- this is necessary?  What makes NumPy arrays (by default?) non-writeable?

Reply:
NumPy arrays are not, by default, non-writeable, but many NumPy operations produce non-writeable arrays. For example, [numpy.diagonal](https://numpy.org/doc/1.18/reference/generated/numpy.diagonal.html#numpy.diagonal) returns a read-only view. And in this case [broadcast_to](https://numpy.org/doc/1.18/reference/generated/numpy.broadcast_to.html?highlight=broadcast_to#numpy.broadcast_to) returns a non-writeable view.

Reply:
ah that's neat.  Although I don't know why we even use a NumPy array here...